### PR TITLE
media-sound/supercollider: add new package

### PR DESCRIPTION
**Pull request by mrVanDalo**
SuperCollider is a platform for audio synthesis and algorithmic composition, used by musicians, artists, and researchers working with sound. It is free and open source software available for Windows, Mac OS X, and Linux.

I created a [Bug Ticket](https://bugs.gentoo.org/show_bug.cgi?id=604096)

### CODE REVIEW

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	^^ ( jack portaudio )"
+
+# Both alsa and readline will be automatically checked in cmake but
+# there are no options for theese. Thus the functionality cannot be
+# controlled through USE flags. Therefore hard-enabled.
+RDEPEND="
+	portaudio? ( media-libs/portaudio )
+	jack? ( >=media-sound/jack2-1.9.10 )
+	media-libs/alsa-lib
+	>=sys-libs/readline-5.0:*
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: `sys-libs/readline:0=`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	x11-libs/libXt
+	avahi? ( net-dns/avahi )
+	fftw? ( >=sci-libs/fftw-3.0 )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: `fftw? ( sci-libs/fftw:3.0= )`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	qt5? (
+		>=dev-qt/qtcore-5.6
+		>=dev-qt/qtwebkit-5.6
+		>=dev-qt/qtsensors-5.6
+		>=dev-qt/qtpositioning-5.6
+		>=dev-qt/qtgui-5.6
+	)
+	sndfile? ( >=media-libs/libsndfile-1.0.16 )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: drop the version specifier, 1.0.16 is long gone

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	system-boost? ( dev-libs/boost )
+	system-yaml? ( dev-cpp/yaml-cpp )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: don't do this - get rid of the `system-boost` and `system-yaml` flags. Also, boost needs to be `dev-libs/boost:=`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+DEPEND="${RDEPEND}
+	dev-libs/icu
+	virtual/pkgconfig
+	emacs? ( virtual/emacs )
+	gedit? ( app-editors/gedit )
+	vim? ( app-editors/vim )"
+
+S="${WORKDIR}/SuperCollider-Source"
+
+DOCS=( AUTHORS README_LINUX.md )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: get rid of these, covered by default set in `einstalldocs`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+		-DCURL=$(usex curl)
+		-DINSTALL_HELP=$(usex doc)
+		-DFFT_GREEN=$(usex !fftw)
+		-DNO_GPL3=$(usex !gpl3)
+		-DNO_SNDFILE=$(usex !sndfile)
+		-DSC_QT=$(usex qt5)
+		-DSCLANG_SERVER=$(usex server)
+		-DLIBSCSYNTH=$(usex !static-libs)
+		-DSYSTEM_BOOST=$(usex system-boost)
+		-DSYSTEM_YAMLCPP=$(usex system-yaml)
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: as mentioned earlier, force both of these on

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+		-DSC_ED=$(usex gedit)
+		-DSC_VIM=$(usex vim)
+		-DSC_EL=$(usex emacs)
+		-DSC_WII=$(usex wiimote)
+	)
+	use jack && mycmakeargs+=( "AUDIOAPI=jack" )
+	use portaudio && mycmakeargs+=( "AUDIOAPI=portaudio" )
+
+	# optimize for machine (will not run on other machines)
+	mycmakeargs+=( -DNATIVE=ON )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: why not attach this to the big list?

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+		-DSSE=$(usex cpu_flags_x86_sse)
+		-DSSE2=$(usex cpu_flags_x86_sse2)
+		-DSC_IDE=$(usex ide)
+		-DSC_ED=$(usex gedit)
+		-DSC_VIM=$(usex vim)
+		-DSC_EL=$(usex emacs)
+		-DSC_WII=$(usex wiimote)
+	)
+	use jack && mycmakeargs+=( "AUDIOAPI=jack" )
+	use portaudio && mycmakeargs+=( "AUDIOAPI=portaudio" )
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: seeing that you have a `^^ REQUIRED_USE`, just make this `AUDIOAPI=$(usex jack jack portaudio)`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+	# optimize for machine (will not run on other machines)
+	mycmakeargs+=( -DNATIVE=ON )
+
+	if [[ $(use debug) ]]
+	then
+		CMAKE_BUILD_TYPE="Debug"
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: in general, don't change `CMAKE_BUILD_TYPE` ever really

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+		-DSC_EL=$(usex emacs)
+		-DSC_WII=$(usex wiimote)
+	)
+	use jack && mycmakeargs+=( "AUDIOAPI=jack" )
+	use portaudio && mycmakeargs+=( "AUDIOAPI=portaudio" )
+
+	# optimize for machine (will not run on other machines)
+	mycmakeargs+=( -DNATIVE=ON )
+
+	if [[ $(use debug) ]]
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: `if use debug; then`

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+	use emacs && newdoc editors/scel/README.md  README.emacs
+	use gedit && newdoc editors/sced/README.md  README.gedit
+}
+
+pkg_postinst() {
+	einfo
+	einfo "Notice: SuperCollider is not very intuitive to get up and running."
+	einfo "The best course of action to make sure that the installation was"
+	einfo "successful and get you started with using SuperCollider is to take"
+	einfo "a look through /usr/share/doc/${PF}/README.gentoo"
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: `einfo "a look through ${EROOT%/}/usr/share/doc/${PF}/README.gentoo"`

Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: where is this `README.gentoo` ever generated? Missing `readme.gentoo-r1.eclass` perhaps?

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 12]
```diff
@@ -0,0 +1,116 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+# are all of theses needed ? 
+inherit cmake-utils eutils multilib versionator
```
Comment (SoapGentoo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: I don't see where you need `eutils`, `multilib` or `versionator`. You will probably need `readme.gentoo-r1` though

**Comment by mrVanDalo** [comment]
@SoapGentoo Thanks for your detailed comments, will fix that tonight

File: media-sound/supercollider/supercollider-3.8.0.ebuild (hunk 7 above)
Comment (mrVanDalo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: I don't know why I did this anymore. Will put this in the big-list

File: media-sound/supercollider/supercollider-3.8.0.ebuild (hunk 6 above)
Comment (mrVanDalo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: I tried but it looks like these two options make the build fail.

**Comment by mrVanDalo** [comment]
@SoapGentoo Sorry that it took me so long, but I had problems to build after forcing boost and yamlccp. Figuring that out took me a while. But I hope everything is fine now.

**Comment by mrVanDalo** [comment]
@SoapGentoo ping.

File: media-sound/supercollider/supercollider-3.8.0.ebuild [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
+
+IUSE="
+	avahi debug emacs +fftw gedit +gpl3 ide jack portaudio qt5 server +sndfile
+	static-libs vim wiimote
+	cpu_flags_x86_sse cpu_flags_x86_sse2"
+
+REQUIRED_USE="
+	ide? ( qt5 )
```
Comment (a17r) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: Is there good reason to have this? As far as I can see, USE=ide does not add another dependency, so why not drop that flag and conflict and let `SC_IDE` be controlled by qt5 as well?

Comment (mrVanDalo) [inline media-sound/supercollider/supercollider-3.8.0.ebuild]: good point. will do it tomorrow in the train.

**Comment by SoapGentoo** [comment]
@mrVanDalo this one is on me, I slacked off a bit :sob:
I changed a number of cosmetic things, please have a look

**Comment by mrVanDalo** [comment]
I see, thanks. No worries :smile:

**Comment by SoapGentoo** [comment]
@mrVanDalo keep the contributions trickling in, Gentoo also thrives through user contributions :smiley:

**Comment by mrVanDalo** [comment]
sure I will :smile: there are some new builds in the line.

**Pull request closed without merging**